src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -0,0 +1,539 @@
+                        //
+                        //  {1 (D "String" (N "System" 0 (N "" 0 (U (S "netstandard" 4) 3) 2) 1) 0 0 (% 0) 0)}
+                        //  {1 (D "String" (N "System" 0 (N "" 0 (U (S "mscorlib" 4) 3) 2) 1) 0 0 (% 0) 0)}
+                        //
+                        // Also we don't use the "ignoreAssemblyKey" option for SymbolKey resolution because its perfermance doesn't meet our requirement.
+                        continue;
+                    }
+
+                    if (CheckedReceiverTypes.TryGetValue(declaredReceiverTypeInCurrentCompilation, out var cachedResult) && !cachedResult)
+                    {
+                        // If we already checked an extension method with same receiver type before, and we know it can't be applied
+                        // to the receiverTypeSymbol, then no need to proceed methods from this group..
+                        continue;
+                    }
+
+                    // This is also affected by the symbol resolving issue mentioned above, which means in case referenced projects
+                    // are targeting different framework, we will miss extension methods with any framework type in their signature from those projects.
+                    var isFirstMethod = true;
+                    foreach (var methodInCurrentCompilation in methodSymbols.Select(s => SymbolFinder.FindSimilarSymbols(s, SemanticModel.Compilation).FirstOrDefault()).WhereNotNull())
+                    {
+                        if (isFirstMethod)
+                        {
+                            isFirstMethod = false;
+
+                            // We haven't seen this receiver type yet. Try to check by reducing one extension method
+                            // to the given receiver type and save the result.
+                            if (!cachedResult)
+                            {
+                                // If this is the first symbol we retrived from current compilation,
+                                // try to check if we can apply it to given receiver type, and save result to our cache.
+                                // Since method symbols are grouped by their declared receiver type, they are either all matches to the receiver type
+                                // or all mismatches. So we only need to call ReduceExtensionMethod on one of them.
+                                var reducedMethodSymbol = methodInCurrentCompilation.ReduceExtensionMethod(ReceiverTypeSymbol);
+                                cachedResult = reducedMethodSymbol != null;
+                                CheckedReceiverTypes[declaredReceiverTypeInCurrentCompilation] = cachedResult;
+
+                                // Now, cachedResult being false means method doesn't match the receiver type,
+                                // stop processing methods from this group.
```

Comment:
is this always true? i.e. what about complex methosd where the reciver type between teh methods might be different.  maybe one doesn't match, but hte others might?

Reply:
Yes I believe so, since we are dealing with actual type symbols here. If two methods' declared receiver type symbols are identical, then one of them matches means the other one does too.